apps/sms/views/conversation/js/conversation.js

```diff
@@ -1159,7 +1165,8 @@ var ConversationView = {
       // If we're leaving a thread's message view,
       // ensure that the thread object's unreadCount
       // value is current (set = 0)
-      if (Threads.active) {
+      if (this.currentThread) {
+        // TODO: Will need to replace with Conversation service method.
```

Comment:
previously we did that but then we decided to move it to after rendering so that rendering coud be as fast as possible.

All this code has been modified billion times :) maybe we can rationalize it a little more